ycmd/handlers.py

```diff
@@ -89,17 +90,39 @@ def RunCompleterCommand():
 def GetCompletions():
   _logger.info( 'Received completion request' )
   request_data = RequestWrap( request.json )
-  do_filetype_completion = _server_state.ShouldUseFiletypeCompleter(
-    request_data )
+  ( do_filetype_completion, forced_filetype_completion ) = (
+                    _server_state.ShouldUseFiletypeCompleter(request_data ) )
   _logger.debug( 'Using filetype completion: %s', do_filetype_completion )
-  filetypes = request_data[ 'filetypes' ]
-  completer = ( _server_state.GetFiletypeCompleter( filetypes ) if
-                do_filetype_completion else
-                _server_state.GetGeneralCompleter() )
 
-  return _JsonResponse( BuildCompletionResponse(
-      completer.ComputeCandidates( request_data ),
-      request_data.CompletionStartColumn() ) )
+  general_completer = _server_state.GetGeneralCompleter()
+  kw = {}
```

Comment:
`kw` is a bit short. I can't tell at-a-glance what this variable is about which is a solid hint that it should be longer.